.gitlab-ci.yml

```diff
@@ -58,6 +58,26 @@ variables:
   environment:
     name: parity-build
 
+.build_to_own_reg:                 &docker_build_to_own_reg
+  <<:                              *docker_build
+  before_script:
+    - test -z "$DOCKERIMAGE" && echo "DOCKERIMAGE must be defined" && exit 1
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
+  script:
+    # $CI_REGISTRY_IMAGE == registry.parity.io/parity/infrastructure/scripts/
+    # $CI_REGISTRY == registry.parity.io/
```

Comment:
where are these variables defined then?

Reply:
two lines below :)

Reply:
I left those comments there just for the information.

Reply:
but don't they have to be defined somewhere - the url registry.parity.io could also be different.

Reply:
nope, `CI_REGISTRY` is the gitlab.parity.io wide and equals registry.parity.io 
and `CI_REGISTRY_IMAGE` will always be `CI_REGISTRY` + current project url

Reply:
so `registry.parity.io` is defined as root registry in gitlab's `/etc/gitlab/gitlab.rb`